test/hw1/test_task4.py

```diff
@@ -0,0 +1,43 @@
+from All_home_works.hw1.task4 import find_maximal_subarray_sum
+
+
+def test_find_maximal_subarray_sum__init_data__end_subarray():
+    nums = [1, 3, -1, -3, 5, 3, 6, 7]
+    k = 3
+    result = 16
+    assert result == find_maximal_subarray_sum(nums, k)
+
+
+def test_find_maximal_subarray_sum__middle_subarray():
```

Comment:
You definitely have your own naming convention for that tests.  This approach might be used when you work closely with the team and everyone are on the same page.  In that case, you can try to use some common English phrases, so readers can understand them without additional context. Just adding a couple of new words to the test name make it easier to understand. 

You can try to build an English phrase and then put it to a test name.

- test_find_maximal_subarray_sum__subarray_in_the_middle
- test_subarray_in_the_middle